src/shogun/machine/KernelMachine.cpp

```diff
@@ -486,7 +486,10 @@ bool CKernelMachine::train_locked(SGVector<index_t> indices)
 		SG_ERROR("CKernelMachine::train_locked() call data_lock() before!\n")
 
 	/* this is asusmed here */
-	ASSERT(m_custom_kernel==kernel)
+	//ASSERT(m_custom_kernel==kernel)
+	SG_UNREF(kernel);
+	kernel=m_custom_kernel;
```

Comment:
clone creates different copy for each.
And since clone happens after data lock, the assert fails. Had to do this.